services/git_service.py

```
@@ -161,9 +164,52 @@ def commit(self, message=None):
                 return False, "没有需要提交的改动"
             return False, f"提交失败: {stderr}"
 
+    def _remote_head(self, remote, branch):
+        """尽力获取 remote-tracking 分支的 HEAD SHA；失败或不存在时返回空串。"""
```

Ruff (0.15.17)

[warning] Docstring contains ambiguous `；` (FULLWIDTH SEMICOLON). Did you mean `;` (SEMICOLON)?

(RUF002)